lib/rack/tracker.rb

```diff
@@ -55,7 +55,11 @@ def html?; @headers['Content-Type'] =~ /html/; end
 
     def inject(env, response)
       @handlers.each(env) do |handler|
-        response.gsub!(%r{</#{handler.position}>}, handler.render + "</#{handler.position}>")
+        if handler.container_position == :closing
+          response.gsub!(%r{</#{handler.container_tag}>}, handler.render + "</#{handler.container_tag}>")
```

Comment:
this inject method gets the response, changes it, and returns it.
If we want to move this inject into the handlers, then we would need to pass the response again to an inject method inside each handler. Doesn't that make it slower and more difficult to understand what is happening? It would also allow the handler to change whatever else it likes in the response!

Another way I could think of doing this is to group handlers according to their container_tag and container_position, render them together, and then gsub once for each group. What do you think about that?